AutoMaxLair.py

```diff
@@ -113,19 +121,19 @@ def detect(inst) -> str:
     # This function returns directly when those conditions are found.
     while True:
         text = inst.read_text(inst.get_frame(), ((0, 0.6), (1, 1)), invert=True)
-        if inst.phrases['FIGHT'] in text:
+        if re.search(inst.phrases['FIGHT'], text) != None:

```

Comment:
Is it necessary to include `!= None`? I think you can omit that part and just have `if x:`. Also I think convention is to use `is not` as opposed to `!=` specifically when comparing to `None`

Reply:
Hmm no it wasn't ! Okay so != and == for numbers and is / is not for other comparable stuff ? (knew to python)

Reply:
IIRC it's a bit more nuanced than that. "==" compares by value whereas "is" checks if both sides are references to the same object. For some reason convention is to use "is" when comparing to None; this approach works because all instances of None are references to a single object. "in" is something else entirely. It checks whether the value on the left is a subset of the value on the right. In this case it was checking if the phrase was contained in the text. Another use case would be to check if the value is found in a list or tuple.